syntax = "proto3";

package flashcards;

option go_package = "memcore/api/proto";

// Message definitions
message CreateTopicRequest {
    int32 user_id = 1;
    int32 topic_id = 2;
    string name = 3;
}

message CreateTopicResponse {
    bool success = 1;
}

message ReviewCompleteRequest {
    int32 user_id = 1;
    int32 card_id = 2;
    int32 rating = 3;
}

message ReviewCompleteResponse {
    int32 next_review_date = 1;
}

message AddCardRequest {
    int32 user_id = 1;
    int32 card_id = 2;
    int32 topic_id = 3;
}

message AddCardResponse {
    bool success = 1;
}

message GetDueCardsRequest {
    int32 user_id = 1;
    int32 date = 2;
    int32 topic_id = 3;
}

message GetDueCardsResponse {
    repeated int32 card_ids = 1;
}

// Service definition
service FlashcardService {
    rpc CreateTopic(CreateTopicRequest) returns (CreateTopicResponse);
    rpc ReviewComplete(ReviewCompleteRequest) returns (ReviewCompleteResponse);
    rpc AddCard(AddCardRequest) returns (AddCardResponse);
    rpc GetDueCards(GetDueCardsRequest) returns (GetDueCardsResponse);
}


